rest_framework/fields.py

```diff
@@ -653,6 +654,30 @@ def to_representation(self, value):
         return str(value)
 
 
+class IPAddressField(CharField):
+    """Support both IPAddressField and GenericIPAddressField"""
+
+    default_error_messages = {
+        'invalid': _('Enter a valid IPv4 or IPv6 address.'),
+    }
+
+    def __init__(self, protocol='both', unpack_ipv4=False, **kwargs):
+        self.protocol = protocol
+        self.unpack_ipv4 = unpack_ipv4
+        super(IPAddressField, self).__init__(**kwargs)
+        validators, error_message = ip_address_validators(protocol, unpack_ipv4)
+        self.validators.extend(validators)
+
+    def to_internal_value(self, data):
+        if data and ':' in data:
+            try:
+                return clean_ipv6_address(data, self.unpack_ipv4)
+            except DjangoValidationError:
+                self.fail('invalid', value=data)
```

Comment:
I'm now back from holidays. I've given a look to the methods. The problem we may have forcing the unpack_ipv4 in case of 'both' is the following:

- the user choose to use 'both' because he/she needs to validate both ipv4 and ipv6.
- the use passes '192.168.0.1' as value, and the value is automatically 'unpacked'.
- the output of 192.168.0.1 is: 2002:c0a8:0001::c0a8:0001

are you sure this is what we want?